man/man8/zfs-receive.8

```diff
@@ -229,6 +229,8 @@ that element to determine the name of the target file system for the new
 snapshot as described in the paragraph above.
 .It Fl h
 Skip the receive of holds.  There is no effect if holds are not sent.
+.It Fl M
+Force an unmount of the file system while receiving a snapshot.
```

Comment:
Unfortunately, this solution isn't going to work for Linux.  The Linux VFS differs from the FreeBSD and Illumos VFS in that it will not allow a filesystem to be unmounted as long as it's in use (even with `MS_FORCE` which means something slightly different).  This behavior is controller by the kernel and isn't something we can change.

What I'd suggest is mentioning in the man page that this option is not support on Linux.  Something like:

```
.It Fl M
Force an unmount of the file system while receiving a snapshot.
This option is not support on Linux.
```